plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php

```diff
@@ -408,20 +406,40 @@ public function get_views() {
 	/**
 	 * Count orders by status.
 	 *
-	 * @param string $status The order status we are interested in.
+	 * @param string|string[] $status The order status we are interested in.
 	 *
 	 * @return int
 	 */
-	private function count_orders_by_status( string $status ): int {
-		$orders = wc_get_orders(
-			array(
-				'limit'  => -1,
-				'return' => 'ids',
-				'status' => $status,
-			)
-		);
+	private function count_orders_by_status( $status ): int {
+		return array_sum( array_map( 'wc_orders_count', (array) $status ) );
```

Comment:
seems like this counts all order types (including refunds, subscriptions etc), although I really like that it does cache out of the box. This implementation is consistent with earlier implementation (where we also were not considering other order type), but the post-type implementation does consider only `shop_order`. (This also makes sense since we will only display only `shop_order` type, so count should be also on only that type). This means we may wanna do the custom query, (or wc_query is also fine, just with specifying types and a limit of 1). WDYT?

Reply:
Thanks for the feedback V!

This function doesn't count refunds nor subscriptions because those order types are registered with [`exclude_from_order_count`](https://github.com/woocommerce/woocommerce/blob/5b1296fe45927cc03b2d286a64e5532afc6640e0/plugins/woocommerce/includes/wc-order-functions.php#L198-L204), which this function [takes into account](https://github.com/woocommerce/woocommerce/blob/5b1296fe45927cc03b2d286a64e5532afc6640e0/plugins/woocommerce/includes/wc-order-functions.php#L359
).

That said, it _could_ end up including other order types if those are not registered with that flag. I'll see if I can adjust `wc_orders_count()` so that we can specify the type, because I thought it'd be best to use an API function instead of relying on a direct query.

That said, if that becomes too cumbersome, I'll default to a direct query or a `WC_Order_Query` as you've suggested.

Reply:
Hey @vedanshujain!

I've tried to implement what I described above. In f8a5fbc0d61ef3e599f4a7a429aaf2a0f7072880, I've added a `$type` argument to `wc_orders_count()` which allows specifying the order type. The arg is optional, and defaults to the same order types as before, so no breakage of backwards compat.

I've also expanded the unit tests related to `wc_orders_count()` (in 3f63a599329dc3c8d8c937e25705eef167b0ec8c) to confirm the new functionality works correctly.

With these changes in place, which might be useful on its own, we can limit the HPOS list table to only `shop_order` types while still using our APIs. In the end, this is mostly the same as a direct query with some benefits.

Please let me know what you think 😸.